src/app/editor/form-generator/components/form-generator/form-generator.component.ts

```diff
@@ -165,9 +165,9 @@ export class FormGeneratorComponent implements OnChanges, AfterViewInit, OnDestr
     /**
      * Resets the isDirty state of the component.
      */
-    setPristine(node: MeshNode): void {
+    setPristine(node: MeshNode, force = false): void {
```

Comment:
Remove the force argument, and instead create a new method:
```TypeScript
update(node: MeshNode): void {
    this.meshControlGroup.checkValue(node.fields as any);
}
```
Then in the NodeEditorComponent you call:
```TypeScript
this.formGenerator.update(node);
this.formGenerator.setPristine(node);
```

This will have the same effect of both updating the values in the form, and setting all fields to be pristine. The advantage of doing it this way is that we retain simple, single-purpose methods.